src/main/scala/org/allenai/pipeline/Pipeline.scala

```diff
@@ -24,23 +27,49 @@ import java.util.Date
 trait Pipeline extends Logging {
   def artifactFactory: FlatArtifactFactory[String] with StructuredArtifactFactory[String]
 
-  def run(rawTitle: String, outputs: Producer[_]*): Unit = {
-    try {
-      val outputInfo = outputs.map(_.stepInfo)
-      require(
-        outputInfo.forall(_.outputLocation.isDefined),
-        s"Running a pipeline without persisting the output: ${
-          outputInfo.filter(_.outputLocation.isEmpty)
-            .map(_.className).mkString(",")
-        }"
-      )
+  private[this] val persistedSteps: ListBuffer[Producer[_]] = ListBuffer()
+
+  /** Run the pipeline.  All steps that have been persisted will be computed, along with any upstream dependencies */
+  def run(title: String): Unit = {
+    run(title, persistedSteps.toSeq)
+  }
 
+  /** Run only specified steps in the pipeline.  Upstream dependencies must exist already.  They will not be computed */
+  def runOnly(title: String, runOnlyTargets: Producer[_]*) {
```

Comment:
If we want to enable the command line flow, we'll need a version that references datasets by name: 
matching.optionallyPersistRdd("matching")
...
pipeline.runOnly(args(0))  // of course, using proper scala options.
...
$ sbt "project pipeline" "runMain BuildIndex --runOnly matching

If we use name reference, it's